### PR TITLE
hive: Fix hive hook output and move lifecycle to cell package

### PR DESCRIPTION
**Pull request by joamaki**
Hi absolutely everyone!

This fixes the hook output that you can see when you run `go run ./daemon hive`
by fixing the "shadowing" caused by `hive.internalLifecycle` and additionally adding module ID to the output to differentiate repeated uses of same start/stop hooks (see below). 

The reason why you got tagged here was because `hive.Lifecycle` is now `cell.Lifecycle` and I did a global rename. Thanks!

Commits:

```
    hive: Remove lifecycle type aliases and refactor uses
    
    This removes the now unnecessary type aliases for lifecycle
    related types from the hive package (hive.Lifecycle etc.)
    and refactors the uses to use cell.Lifecycle etc.
```

```
    hive: Add module ID to hive start/stop hook output
    
    The move of the lifecycle into separate package and having two
    implementations caused the hook output to be broken. This moves the
    lifecycle to the cell package and removes the internalLifecycle.
    With the lifecycle in the cell package we then add module IDs to the
    hooks to provide where the hook was added from as there's already many
    duplicate start/stop functions (e.g. from health reporter and job groups).
    
    "go run ./daemon hive" before:
    
        • hive.(*internalLifecycle).Append.func2 (pkg/hive/hive.go:185)
        • *job.group.Stop
    
    After:
    
        • *cell.reporterHooks.Stop (agent.datapath.bandwidth-manager)
        • *job.group.Stop (agent.datapath.l2-responder)
    
    Fixes: 7f94c897a1 ("hive: move lifecycle types to separate package.")
```

### CODE REVIEW

**Comment by joamaki** [comment]
Hmm, now I'm actually having second thoughts about removing the type aliases as this impacts both backporting and forks of Cilium. OTOH, would be nice to consolidate and it's super easy to fix the issues. Let me know what you think.

**Review by dylandreimerink — APPROVED** [review]

**Review by lmb — APPROVED** [review]
No opinion re the aliases. If you don't make the change now it'll probably never happen.

**Review by giorio94 — APPROVED** [review]
/lgtm for my codeowners

**Review by pippolo84 — APPROVED** [review]

**Review by rastislavs — APPROVED** [review]

**Review by squeed — APPROVED** [review]

**Review by chancez — APPROVED** [review]

File: pkg/hive/cell/module.go
Comment (tklauser) [inline pkg/hive/cell/module.go]: Leftover debug message?

Comment (joamaki) [reply pkg/hive/cell/module.go]: Oopsie, fixed!

**Review by ghost — APPROVED** [review]
docs good

**Review by nathanjsweet — APPROVED** [review]

**Review by sayboras — APPROVED** [review]
LGTM :heavy_check_mark:

**Review by pchaigno — APPROVED** [review]
Approving so GH is happy.

**Review by julianwiedmann — APPROVED** [review]

**Comment by joestringer** [comment]
Definitely something weird going on, but I suspect that once someone from @sig-ipam approves it should go through.

**Review by tklauser — APPROVED** [review]

**Pull request merged**
